### PR TITLE
Implemented react tutorial required steps

### PR DESCRIPTION
**Pull request by rskinnerc**
#### In this PR I have: 

- Developed the following parts:
[Part 1 - React tutorial: The beginner's guide to learning React in 2020](https://ibaslogic.com/react-tutorial-for-beginners/).
[Part 2 - Working with React form and handling event](https://ibaslogic.com/react-form-handling/).
[Part 3 – How to implement CSS in Reactjs app](https://ibaslogic.com/css-in-reactjs-app/).
[Part 4 – How to edit to dos items](https://ibaslogic.com/how-to-edit-todos-items-in-react/).
[Part 5 – Persisting React state in local storage](https://ibaslogic.com/persisting-react-state-in-local-storage/) (Optional).
[Part 6 – Getting started with React lifecycle methods](https://ibaslogic.com/react-lifecycle-methods/).
[Part 7 – Getting started with React hooks](https://ibaslogic.com/react-hooks-tutorial/).
[Part 8 – How to use SVG icons in React NOT IMPLEMENTED](https://ibaslogic.com/how-to-use-svg-icons-in-react-project/) (Optional).
[Part 9 – Routing with React Router](https://ibaslogic.com/routing-with-react-router/).
[Part 10 – How to add hamburger menu in React NOT IMPLEMENTED](https://ibaslogic.com/how-to-add-hamburger-menu-in-react/) (Optional).
[Part 11 – Deploying React app to GitHub Pages](https://ibaslogic.com/deploying-react-app-to-github-pages/) (Optional).

- Installed linters and fixed linter errors
- Deployed the app - [See Live](https://deploy-preview-1--loquacious-marigold-fb89e2.netlify.app/)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import TodoContainer from './components/TodoContainer';
+
+import './App.css';
+import About from './pages/About';
+import NoMatch from './pages/NoMatch';
+import Navbar from './components/Navbar';
+import SinglePage from './pages/SInglePage';
+
+ReactDOM.render(
```
Comment (devMahmoud) [inline src/index.js:12]: - [ ] You should use createRoot instead of ReactDOM.render

dom

File: .gitignore [hunk 2]
```diff
@@ -1 +1,23 @@
-.vscode
\ No newline at end of file
+.vscode# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (devMahmoud) [inline .gitignore:1]: - [ ] .vscode folder didn't get exclude try to fix this problem

Screenshot from 2022-05-31 02-24-56

**Review by devMahmoud — CHANGES_REQUESTED** [review]
## Changes required :recycle: 

Hi, @rskinnerc 
Good job! 

quality-quality-work

Your code is clean and everything is working fine but there are small changes that could make it even better

## (Highlights) Good Points: :+1: 
* You documented your work professionally. :heavy_check_mark:
* Gitflow is used correctly. :heavy_check_mark:
* All linters checks are ok. :heavy_check_mark:

## [Optional] suggestions:

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding! :clap::clap::clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

File: src/index.js (hunk 1 above)
Comment (rskinnerc) [reply src/index.js]: @devMahmoud Solved. Thank you for your feedback

File: .gitignore (hunk 2 above)
Comment (rskinnerc) [reply .gitignore]: @devMahmoud solved. Thank you for your feedback

**Review by devMahmoud — APPROVED** [review]
## Approved

Hi, @rskinnerc 

Good job! your work is complete there is nothing else to say other than it's time to merge it.

8tuI


## (Highlights) Good Points: :+1: 
* You documented your work professionally. :heavy_check_mark:
* Gitflow is used correctly. :heavy_check_mark:
* All linters checks are ok. :heavy_check_mark:

## [Optional] suggestions:

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding! :clap::clap::clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

**Pull request merged**